### PR TITLE
[WIP] Guides

### PR DESCRIPTION
**Pull request by solnic**
### Summary

This is outstanding work of @nepalez on Mapper guides. This PR does a couple of things too:

* doc-pages are now generated dynamically so we only need to add new doc pages to `doc-pages` dir and things will happen automatically
* new Guides section is added to the main navigation
* Guides include `Mappers` section with a ton of fresh content

### TODO

- [ ] Write introduction to Guides
- [ ] wrap up (no pun intended) guides for mappers
- [ ] add sub-navigation and ToC for docs

### CODE REVIEW

**Comment by maetl** [comment]
> doc-pages are now generated dynamically so we only need to add new doc pages to doc-pages dir and things will happen automatically

Thanks!

File: source/doc-pages/guides.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Guides
+
+TODO: write intro to guides
```
Comment (maetl) [inline source/doc-pages/guides.md]: I can take care of this bit.

Comment (solnic) [inline source/doc-pages/guides.md]: It's all yours :) I'm gonna add a TODO to the description of this PR

File: source/doc-pages/tutorials/todo-app-with-sinatra.md [hunk 2]
```diff
@@ -1,232 +0,0 @@
-In this tutorial, we’ll walk through the steps required to build a web API for managing a todo list.
-
-Each example builds on the last, showing how ROM and Sinatra can work together to provide an effective toolkit for rapid prototyping microservices and REST APIs. 
-
-For simplicity and clarity, this tutorial uses ROM’s setup DSL, though in larger apps, we recommend configuring ROM with explicit class definitions. This is analogous to the difference between Sinatra’s routing DSL and its modular application style.
```
Comment (maetl) [inline source/doc-pages/tutorials/todo-app-with-sinatra.md]: @nepalez ^ see the above paragraph for at least one take on the differences/use of the ROM setup DSL.

Comment (nepalez) [inline source/doc-pages/tutorials/todo-app-with-sinatra.md]: Thanks! Will use it

**Comment by solnic** [comment]
What do you think is essential to finish so that we could publish first version of the guides? I'd love to release it next week along with ROM 0.8.0 :)

**Comment by nepalez** [comment]
IMO, there is in descending importance:

* finish glossary
* provide the adapter guide
* get though all guides by native English speaker
* create various appendixes for mappers (they are still empty instead of `wrapping`)

Unfortunately, I stick in `ungroup`/`unfold` features and haven't written much this week.

**Comment by solnic** [comment]
Thanks @nepalez. I'm gonna work on finishing glossary and adding the adapter guide. I'll also update command section with info about new graph feature that's coming in 0.8.0.

File: source/doc-pages/guides/basics/relations.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+``` ruby
+class Users < ROM::Relation[:memory]
+end
+```
+
+Alternatively you can use the routing-style DSL:
+
+```ruby
+ROM.relation(:users) do
```
Comment (cflipse) [inline source/doc-pages/guides/basics/relations.md]: Might want to show how you specify the memory adapter here, so it's a more clear parallel to the class definition above.

**Comment by solnic** [comment]
This is now merged. Thanks <3

**Pull request merged**